bebraven/platform
# Grading rake task

Task: https://app.asana.com/0/0/1189791588110885/f

Summary: Using the existing lesson interactions, grade calculator, and canvas API, add a rake task that ties them all together and sends grades to canvas.

Screenshot with 50% on each of three mastery quizzes, plus 100% module completion, `(50+100)/200 = 75`:
<img width="881" alt="Screen Shot 2020-09-03 at 10 24 57 AM" src="https://user-images.githubusercontent.com/1382374/92134963-c8bc0580-edcf-11ea-8bbe-f8db6baf8290.png">

Testing:

1. Upload a new Rise 360 module as a Canvas assignment, worth 10 points.
2. Log into canvas as a student and complete the module.
3. Run the rake task, `docker-compose exec platformweb bundle exec rake grade:lessons`
4. Check the grade tab for the correct grade in that assignment.

Example honeycomb trace: https://ui.honeycomb.io/braven/datasets/development/result/ga8nKtLSvPQ/trace/xsPpFiMZZQy
